File: 02/AED2/TrabalhoTeorico02_ProcessoAEDsII/02_DonaMonica/DonaMonica.c
#include <stdio.h>
#include <string.h>

int getAge(int x, int y, int z){
    if (x > y && x > z)
        return x;
    else if (y > x && y > z)
        return y;
    return z;
}

int calcAge(int m, int x, int y){
    return m - (x + y);
}

int main(char **args){
    char line[6];
    int m, x, y, z;

    while (1)
    {
        scanf("%d", &m);

        if (m == 0)
            return 1;

        scanf("%d %d", &x, &y);

        z = calcAge(m, x, y);
        printf("%d\n", getAge(x, y, z));
    }

    return 1;
}

File: 02/AED2/TrabalhoTeorico02_ProcessoAEDsII/02_DonaMonica/DonaMonica.java
import java.util.Scanner;

public class DonaMonica {
    private static int getAge(int x, int y, int z) {
        if (x > y && x > z)
            return x;
        else if (y > x && y > z)
            return y;

        return z;
    }

    private static int calcAge(int m, int x, int y) {
        return (m - (x + y));
    }

    public static void main(String[] args){
        Scanner s = new Scanner(System.in);

        int m, x, y, z;

        while(true){

            m = s.nextInt();
            
            if (m == 0)
                break;

            x = s.nextInt();
            y = s.nextInt();
        
            z = calcAge(m,x,y);
        System.out.println(getAge(x,y,z));
        }
    }
}
